presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -128,6 +129,18 @@ protected void parseURI(String requestURI, HttpServletRequest request, HttpServl
         OutputBufferId bufferId = null;
         long token = 0;
 
+        Enumeration<String> headerNames = request.getHeaderNames();
+        while (headerNames.hasMoreElements()) {
```

Comment:
I think it would be better to not throw if the header contains a "\r" or "\n". Instead remove these characters, and proceed with the sanitized header string.

Reply:
`HttpServletRequest` is read only. I can only make a new request wrapper that returns the header values that have "\r" and "\n" removed. Not sure this is actually what we want.

Reply:
[According to the grammar in RFC 7230](https://www.rfc-editor.org/rfc/rfc7230#section-3.2.6) `\r` and `\n` are not valid characters at all in header field names or in header values. I think it is fair to just throw an exception. It will notify the upstream request/proxy they've done something invalid to the request.